lib/GeometryMath.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+var Cesium = require('cesium');
+var Cartesian3 = Cesium.Cartesian3;
+var defined = Cesium.defined;
+var DeveloperError = Cesium.DeveloperError;
+
+module.exports = {
+    sumBarycentric: sumBarycentric,
+    flattenTriangle: flattenTriangle,
+    pointLineDistanceParametric: pointLineDistanceParametric
+};
+
+// Functions for working with geometry like triangles and lines
+
+var sumBarycentricScratch = new Cartesian3();
+function sumBarycentric(barycentric, vector0, vector1, vector2, result) {
+    //>>includeStart('debug', pragmas.debug);
+    if (!defined(barycentric)) {
+        throw new DeveloperError('barycentric is required');
+    }
+    if (!defined(vector0)) {
+        throw new DeveloperError('vector0 is required');
+    }
+    if (!defined(vector1)) {
+        throw new DeveloperError('vector1 is required');
+    }
+    if (!defined(vector2)) {
+        throw new DeveloperError('vector2 is required');
+    }
+    if (!defined(result)) {
+        throw new DeveloperError('result is required');
+    }
+    //>>includeEnd('debug');
+
+    result.x = 0.0;
+    result.y = 0.0;
+    result.z = 0.0;
+    Cartesian3.multiplyByScalar(vector0, barycentric.x, sumBarycentricScratch);
+    Cartesian3.add(result, sumBarycentricScratch, result);
+    Cartesian3.multiplyByScalar(vector1, barycentric.y, sumBarycentricScratch);
+    Cartesian3.add(result, sumBarycentricScratch, result);
+    Cartesian3.multiplyByScalar(vector2, barycentric.z, sumBarycentricScratch);
+    Cartesian3.add(result, sumBarycentricScratch, result);
+    return result;
```

Comment:
I think these lines might actually be clearer by not using the `Cartesian` functions, and probably faster too.